app/graphql/types/category.rb

```diff
@@ -0,0 +1,46 @@
+class Types::Category < Types::BaseObject
+  description 'Information about a specific Category'
+
+  field :id, ID, null: false
+
+  field :title, String,
+    null: false,
+    description: 'The name of the category.'
+
+  field :description, String,
+    null: true,
+    description: 'A description of the category.'
+
+  field :slug, String,
+    null: false,
+    description: 'The URL-friendly identifier of this Category.'
+
+  field :anidb_id, ID,
+    null: true,
+    description: 'Kitsu category mapping to Anidb category.'
+
+  field :total_media_count, Integer,
+    null: false,
+    description: 'The amount of media that include this category.'
+
+  field :nsfw, Boolean,
+    null: false,
+    description: 'Whether the category is Not-Safe-for-Work.',
+    method: :nsfw?
+
+  field :parent, Types::Category,
+    null: true,
+    description: 'The parent category. Each category can have one parent.'
+
+  field :child_count, Integer,
```

Comment:
As above, we should push these counts onto connection objects. For this one you might be able to customize the category connection type already and ditch the counter cache!

Reply:
for child count we are only going 1 level down correct? We wouldn't want to count the children of children.